packages/client/auth/src/CrossmintAuthClient.ts

```diff
@@ -0,0 +1,182 @@
+        return getCookie(SESSION_PREFIX);
+    }
+
+    public async getUser() {
+        const result = await this.apiClient.get(`api/${CROSSMINT_API_VERSION}/sdk/auth/user`, {
+            headers: {
+                "Content-Type": "application/json",
+            },
+        });
+
+        const user = await result.json();
+        return user;
+    }
+
+    public storeAuthMaterial(authMaterial: AuthMaterialWithUser) {
+        setCookie(SESSION_PREFIX, authMaterial.jwt);
+        setCookie(REFRESH_TOKEN_PREFIX, authMaterial.refreshToken.secret, authMaterial.refreshToken.expiresAt);
+    }
+
+    public logout() {
+        deleteCookie(REFRESH_TOKEN_PREFIX);
+        deleteCookie(SESSION_PREFIX);
+        this.callbacks.onLogout?.();
+    }
+
+    public async handleRefreshToken(refreshTokenSecret?: string): Promise<void> {
+        const refreshToken = refreshTokenSecret ?? getCookie(REFRESH_TOKEN_PREFIX);
+        if (refreshToken == null || this.isRefreshing) {
+            return;
+        }
+
+        try {
+            this.isRefreshing = true;
+            const authMaterial = await this.refreshAuthMaterial(refreshToken);
+            this.storeAuthMaterial(authMaterial);
+            this.callbacks.onTokenRefresh?.(authMaterial);
+
+            this.scheduleNextRefresh(authMaterial.jwt);
+        } catch (error) {
+            console.error(error);
+            this.logout();
+        } finally {
+            this.isRefreshing = false;
+        }
+    }
+
+    public async getOAuthUrl(provider: OAuthProvider) {
+        const result = await this.apiClient.get(
+            `api/${CROSSMINT_API_VERSION}/session/sdk/auth/social/${provider}/start`,
+            {
+                headers: {
+                    "Content-Type": "application/json",
+                },
+            }
+        );
+
+        const data = (await result.json()) as { oauthUrl: string };
+        return data.oauthUrl;
+    }
+
+    public async sendEmailOtp(email: string) {
+        const result = await this.apiClient.post(`api/${CROSSMINT_API_VERSION}/session/sdk/auth/otps/send`, {
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({ email }),
+        });
+
+        return await result.json();
+    }
+
+    public async confirmEmailOtp(email: string, emailId: string, token: string) {
+        const queryParams = new URLSearchParams({
+            email,
+            signinAuthenticationMethod: "email",
+            token,
+            locale: "en",
+            state: emailId,
+            callbackUrl: `${this.apiClient.baseUrl}/api/${CROSSMINT_API_VERSION}/session/sdk/auth/we-dont-actually-use-this-anymore`,
+        });
+        const result = await this.apiClient.post(
+            `api/${CROSSMINT_API_VERSION}/session/sdk/auth/authenticate?${queryParams}`,
+            {
+                headers: {
+                    "Content-Type": "application/json",
```

Comment:
nitpick, but it just crossed my mind: why isn't the `apiClient` adding this header for every request? we have non-json endpoints?

Reply:
I thought about this too, but didn't want to touch the `apiClient` just in case and followed the same pattern used elsewhere